solutionbox/code_free_ml/test_mltoolbox/test_transform_raw_data.py

```diff
@@ -36,112 +25,79 @@ class TestTransformRawData(unittest.TestCase):
+    features_file = os.path.join(self.source_dir, 'features.json')
+    file_io.write_string_to_file(features_file, json.dumps(features))
+    cmd = ['python ' + os.path.join(CODE_PATH, 'analyze_data.py'),
+           '--output-dir=' + self.analysis_dir,
+           '--csv-file-pattern=' + self.csv_input_filepath,
+           '--csv-schema-file=' + schema_file,
+           '--features-file=' + features_file]
+    subprocess.check_call(' '.join(cmd), shell=True)
 
   def test_local_csv_transform(self):
     """Test transfrom from local csv files."""
     try:
       self._create_test_data()
-      tfex_dir = os.path.join(self.output_folder, 'test_results')
+
       cmd = ['python ' + os.path.join(CODE_PATH, 'transform_raw_data.py'),
              '--csv-file-pattern=' + self.csv_input_filepath,
-             '--analyze-output-dir=' + self.output_folder,
+             '--analyze-output-dir=' + self.analysis_dir,
              '--output-filename-prefix=features',
-             '--output-dir=' + tfex_dir]
+             '--output-dir=' + self.output_dir]
       subprocess.check_call(' '.join(cmd), shell=True)
 
       # Read the tf record file. There should only be one file.
-      record_filepath = os.path.join(tfex_dir,
+      record_filepath = os.path.join(self.output_dir,
                                      'features-00000-of-00001.tfrecord.gz')
       options = tf.python_io.TFRecordOptions(
           compression_type=tf.python_io.TFRecordCompressionType.GZIP)
-      serialized_example = next(
-          tf.python_io.tf_record_iterator(
-              record_filepath,
-              options=options))
+      serialized_examples = list(tf.python_io.tf_record_iterator(record_filepath, options=options))
+      self.assertEqual(len(serialized_examples), 3)
+
       example = tf.train.Example()
-      example.ParseFromString(serialized_example)
+      example.ParseFromString(serialized_examples[0])
 
       transformed_number = example.features.feature['num_col'].float_list.value[0]
-      self.assertAlmostEqual(transformed_number, 24.0)
-
-      image_bytes = example.features.feature['img_col'].bytes_list.value[0]
-      raw_img = Image.open(self.img_filepath).convert('RGB')
-      img_file = six.BytesIO()
-      raw_img.save(img_file, 'jpeg')
-      expected_image_bytes = img_file.getvalue()
-
-      self.assertEqual(image_bytes, expected_image_bytes)
+      self.assertAlmostEqual(transformed_number, 23.0)
+      transformed_category = example.features.feature['cat_col'].int64_list.value[0]
+      self.assertEqual(transformed_category, 2)
+      image_bytes = example.features.feature['img_col'].float_list.value
+      self.assertEqual(len(image_bytes), 2048)
+      self.assertTrue(any(x != 0.0 for x in image_bytes))
 
     finally:
-      shutil.rmtree(self.output_folder)
+      pass
+      shutil.rmtree(self.working_dir)
 
   def test_local_bigquery_transform(self):
     """Test transfrom locally, but the data comes from bigquery."""
```

Comment:
The original thinking was that since image transformer needs to download a checkpoint from GCS anyway, I would test it using the images from the same bucket. But I agree that we should make the test self contained as much as possible. So I made the change to create images instead of downloading from GCS.